AdminDashboard/Validators/Accounts/AccountValidator.cs

```diff
@@ -0,0 +1,35 @@
+﻿using AdminDashboard.Repositories.AccountRepository;
+using System;
+using System.Threading.Tasks;
+
+namespace AdminDashboard.Validators.Accounts
+{
+    public class AccountValidator : IAccountValidator
+    {
+        private readonly IAccountRepository Repository;
+
+        public AccountValidator(IAccountRepository repository)
+        {
+            Repository = repository;
+        }
+
+        public async Task<bool> DoesAccountExist(Guid id)
+        {
+            return await Repository.GetAccountBy(id) == null ? false : true;
+        }
+
+        public async Task<bool> DoesAccountAlreadyHaveUser(Guid accountId, Guid userId)
+        {
+            var account = await Repository.GetAccountBy(accountId);
+
+            return account.Users.Exists(u => u == userId);
+        }
+
+        public async Task<bool> HasAccountReachedMaximumNumberOfUsers(Guid accountId)
```

Comment:
What's the purpose of this AccountValidator? Since it's used in HTTP handlers, I would assume that this is to validate input parameters but it's not actually the case here.